.github/workflows/development.yml

```diff
@@ -0,0 +1,56 @@
+name: Development Test & Deploy
+run-name: ${{ github.ref_name }} (dev)
+on:
+  push
+permissions: {}
```

Comment:
This sets permissions to none, which it turns out is fine for almost everything. The required permissions can then be granted on a per job basis.